### DISCUSSION TITLE
How to delete all assets in the collection?

### DISCUSSION THREAD

**Discussion by tomekit** [isar/isar#1475] [Help] [answered]
In the documentation: https://isar.dev/crud.html#modifying-the-database there is an example

```
final count = await isar.recipes.filter().isFavoriteEqualTo(false).deleteAll();
```

Is there any way to `deleteAll` without specifying any condition?

**Comment by Jtplouffe** [comment, marked as answer]
You should be able to call either `clear()` or `where().deleteAll()`, like so:
```dart
await isar.writeTxn(() async {
  await isar.recipes.clear();
  // OR
  await isar.recipes.where().deleteAll();
});
```

The difference between the two is that `clear` will also reset the auto increment id.

**Reply by arnab** [reply]
Is `clear` faster than `where.deleteAll` for large amounts of data (say thousands of records)?
